rcl/test/rcl/test_info_by_topic.cpp

```diff
@@ -106,12 +107,16 @@ class CLASSNAME (TestInfoByTopicFixture, RMW_IMPLEMENTATION) : public ::testing:
     EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
   }
 
-  void assert_qos_equality(rmw_qos_profile_t qos_profile1, rmw_qos_profile_t qos_profile2)
+  void assert_qos_equality(
+    rmw_qos_profile_t qos_profile1, rmw_qos_profile_t qos_profile2,
+    bool is_publisher)
   {
     EXPECT_EQ(qos_profile1.deadline.sec, qos_profile2.deadline.sec);
     EXPECT_EQ(qos_profile1.deadline.nsec, qos_profile2.deadline.nsec);
-    EXPECT_EQ(qos_profile1.lifespan.sec, qos_profile2.lifespan.sec);
-    EXPECT_EQ(qos_profile1.lifespan.nsec, qos_profile2.lifespan.nsec);
+    if (is_publisher) {
```

Comment:
why this change?

Reply:
I excluded this check for the subscription because a DDS reader does not support the lifespan qos (see DDS spec section 2.2.3). Getting the lifespan from a subscription (reader) returns the default value infinite, which makes the test fail.

Reply:
Makes sense